fetch/api/request/request-init-contenttype.any.js

```diff
@@ -0,0 +1,139 @@
+function requestFromBody(body) {
+  return new Request(
+    "https://example.com",
+    {
+      method: "POST",
+      body,
+    },
+  );
+}
+
+test(() => {
+  const request = requestFromBody(undefined);
+  assert_equals(request.headers.get("Content-Type"), null);
+}, "Default Content-Type for Request with empty body");
+
+test(() => {
+  const blob = new Blob([]);
+  const request = requestFromBody(blob);
+  assert_equals(request.headers.get("Content-Type"), null);
+}, "Default Content-Type for Request with Blob body (no type set)");
+
+test(() => {
+  const blob = new Blob([], { type: "" });
+  const request = requestFromBody(blob);
+  assert_equals(request.headers.get("Content-Type"), null);
+}, "Default Content-Type for Request with Blob body (empty type)");
+
+test(() => {
+  const blob = new Blob([], { type: "a/b; c=d" });
+  const request = requestFromBody(blob);
+  assert_equals(request.headers.get("Content-Type"), "a/b; c=d");
+}, "Default Content-Type for Request with Blob body (set type)");
```

Comment:
Can you test a non-canonicalized MIME type, e.g. `Text/HTML;Charset="utf-8"`? I can't remember whether this should canonicalize (as `text/html;charset=utf-8`) or be left as-is.

Similarly testing an invalid-but-ASCII MIME type (e.g. `"foo"`) and a Unicode MIME type (e.g. `"😋/😁"`) would be worthwhile IMO.

Reply:
MIME types in `Blob`s are known to be broken, see whatwg/mimesniff#43 and in particular https://github.com/w3c/FileAPI/issues/43#issuecomment-840441099. The general consensus in that issue seems to be that `Blob`s should parse and serialize the MIME type, and some tests were added in #7764, but the spec hasn't been changed yet. So I'd rather not add new tests that contradict the ones from #7764.

Reply:
Fair enough.